### PR TITLE
Biomarker menu

### PR DESCRIPTION
**Pull request by Knight7561**

### CODE REVIEW

File: projects/ccf-database/src/lib/queries/ontology-tree-n3.ts [hunk 1]
```diff
@@ -142,3 +142,34 @@ export const getAnatomicalStructureTreeModel = memoize(getAnatomicalStructureTre
 export function getCellTypeTreeModel(store: Store): OntologyTreeModel {
   return getOntologyTreeModel(store, rui.cell.id, 'cell', ccf.asctb.ct_is_a.id);
 }
+
+function formBiomarkerNode(id,parent,children): OntologyTreeNode {
```
Comment (axdanbol) [inline projects/ccf-database/src/lib/queries/ontology-tree-n3.ts:146]: Missing types for arguments
Also run formatter on file

File: projects/ccf-eui/src/app/modules/ontology-exploration/ontology-tree/ontology-tree.component.html [hunk 2]
```diff
@@ -1,58 +1,73 @@
...
-      (click)="select($event.ctrlKey, node, true, !isSelected(node))"
-    >{{ getNodeLabel(node.label) }}</div>
+      [class.unavailable]="!termData[node.original.id]" [class.selected]="isSelected(node)"
+      (click)="select($event.ctrlKey, node, true, !isSelected(node))">{{ getNodeLabel(node.label) }}</div>
   </ng-template>
 
   <!-- Leaf node template -->
   <mat-tree-node *matTreeNodeDef="let node" class="node leaf-node" matTreeNodePadding
-  [matTreeNodePaddingIndent]="indent" (mouseleave)="mouseOut(); slider1.reset()">
+    [matTreeNodePaddingIndent]="indent" (mouseleave)="mouseOut(); slider1.reset()" style="display: block;">
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/modules/ontology-exploration/ontology-tree/ontology-tree.component.html:14]: Can't this style be moved to the scss file?

File: projects/ccf-eui/src/app/modules/ontology-exploration/ontology-tree/ontology-tree.component.html [hunk 3]
```diff
@@ -1,58 +1,73 @@
...
+  <div class="biomarkers-toggle" *ngIf="showtoggle && node.label==='biomarkers'">
+    <ccf-button-toggle [menuOptions]="menuOptions" [selectedItems]="selectedtoggleOptions"
+      (selectionChange)="toggleSelection($event)"></ccf-button-toggle>
+  </div>
   </mat-tree-node>
 
   <!-- Inner node template -->
   <mat-tree-node *matTreeNodeDef="let node; when: isInnerNode" class="node inner-node" matTreeNodePadding
-                 [matTreeNodePaddingIndent]="indent" (mouseleave)="mouseOut(); slider2.reset()">
+    [matTreeNodePaddingIndent]="indent" (mouseleave)="mouseOut(); slider2.reset()" style="display: block;">
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/modules/ontology-exploration/ontology-tree/ontology-tree.component.html:43]: Can't this style be moved to the scss file?

File: projects/ccf-eui/src/app/shared/components/button-toggle/button-toggle.component.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  /**
+   * Any time a button is clicked, event is emitted.
+   */
+  @Output() readonly selectionChange = new EventEmitter<string[]>();
+
+  isItemSelected(item: string) {
+    return this.selectedItems?.includes(item);
+  }
+
+  toggleSelection(value) {
```
Comment (axdanbol) [inline projects/ccf-eui/src/app/shared/components/button-toggle/button-toggle.component.ts:29]: Missing types

File: projects/ccf-eui/src/styles.scss [hunk 5]
```diff
@@ -11,6 +11,7 @@
 @use 'themes/gtex_dark';
 
 @import 'swiper/swiper.min.css';
+@import "~@angular/material/prebuilt-themes/indigo-pink.css";
```
Comment (axdanbol) [inline projects/ccf-eui/src/styles.scss:14]: This doesn't seem quite right... Why has this been added?

File: projects/ccf-shared/src/lib/services/data-source/api-endpoint.service.ts [hunk 6]
```diff
@@ -131,6 +131,51 @@ export class ApiEndpointDataSourceService implements DataSource {
...
+        'biomarker4' : this.formBiomarkerNode('biomarker4','biomarkers',[],'protein'),
+        'biomarker5' : this.formBiomarkerNode('biomarker5','biomarkers',[],'lipid'),
+        'biomarker6' : this.formBiomarkerNode('biomarker6','biomarkers',[],'lipid'),
+        'biomarkers' : this.formBiomarkerNode('biomarkers','',['biomarker1','biomarker2','biomarker3','biomarker4','biomarker5','biomarker6']),
+      }
+    };
+    return of(a as OntologyTreeModel);
+  }
+
+  // TODO: Use this code when this.api.biomarkerTreeModel is available
```
Comment (axdanbol) [inline projects/ccf-shared/src/lib/services/data-source/api-endpoint.service.ts:170]: @bherr2 Is this not available yet?

**Review by axdanbol — CHANGES_REQUESTED** [review]

File: projects/ccf-shared/src/lib/services/data-source/api-endpoint.service.ts (hunk 6 above)
Comment (bherr2) [reply projects/ccf-shared/src/lib/services/data-source/api-endpoint.service.ts:170]: Correct. I'll add the necessary code but where they put the mock data is good.

File: projects/ccf-eui/src/styles.scss (hunk 5 above)
Comment (Knight7561) [reply projects/ccf-eui/src/styles.scss:14]: The toggle button styles remain consistent with this import. If we remove this import the borders and certain styles are not visible for non-selected buttons.

Comment (bherr2) [reply projects/ccf-eui/src/styles.scss:14]: Then you have a larger problem that you don't understand.

**Review by bherr2 — APPROVED** [review]

**Pull request merged**